zebra-chain/src/block/tests.rs

```diff
@@ -130,6 +133,167 @@ fn deserialize_block() {
         .expect("block test vector should deserialize");
 }
 
+#[test]
+fn block_limits_multi_tx() {
+    // Test multiple small transactions to fill a block max size
+    // A dummy transaction
+    let tx = Transaction::zcash_deserialize(&zebra_test::vectors::DUMMY_TX1[..]).unwrap();
+
+    // A block header
+    let blockheader = generate_block_header();
+
+    // Serialize header
+    let mut data_header = Vec::new();
+    blockheader
+        .zcash_serialize(&mut data_header)
+        .expect("Block header should serialize");
+
+    // Calculate the number of transactions we need
+    let mut max_transactions_in_block =
+        (MAX_BLOCK_BYTES as usize - data_header.len()) / *&zebra_test::vectors::DUMMY_TX1[..].len();
+
+    // Create transactions to be just below the limit
+    let mut many_transactions = std::iter::repeat(Arc::new(tx.clone()))
+        .take(max_transactions_in_block)
+        .collect::<Vec<Arc<_>>>();
```

Comment:
fwiw the `Arc<_>` here can probably be inferred by the compiler
```suggestion
        .collect::<Vec<_>>();
```